haystack/components/joiners/string_joiner.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List
+
+from haystack import component, logging
+from haystack.core.component.types import Variadic
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class StringJoiner:
+    """
+    Component to join strings from different components to a list of strings
```

Comment:
```suggestion
    Component to join strings from different components to a list of strings.
```

Let's also add a simple usage example (you can reuse the one from behavioral test).